### DISCUSSION TITLE
What's the best approach to disable the back/forward browser cache?

### DISCUSSION THREAD

**Discussion by JacobSNGoodwin** [solidjs/solid-router#384] [Q&A]
I'm working on page which requires an authorized user to manage a list of data. When the user logs out, I redirect them to a login page. 

However, if I press the back button, the browser back/forward cache takes over, and displays the last page (see [point 4](https://github.com/solidjs/solid-router?tab=readme-ov-file#cache) in the cache documentation). As far as I can tell from the code, this cache lasts 180000 ms: [cache.ts]( https://github.com/solidjs/solid-router/blob/main/src/data/cache.ts#L16 )

I would prefer that the user is not able to see the content of the page which requires an authorized user. It's not the end of the world, as taking any future action on the authorized data would eventually land them back at a login page. But I still don't like the idea of showing that cached data. 

Is there any mechanism for accomplishing this? Maybe via some client side code and action?

This is an example of a layout for a page using solid-start's FileRouter. I've tried playing around with the `load` function, but am struggling to figure this out. 

[Example Route](https://github.com/JacobSNGoodwin/scavenge-solid/blob/errors-loging/src/routes/manage.tsx)

**Comment by JacobSNGoodwin** [comment]
I'll note that a much easier example is solid-start's own [with-auth example](https://github.com/solidjs/solid-start/tree/main/examples/with-auth), where you can log out, and the click the browser back button to show the page. 

I've tried some some cache header controls and some client-side workarounds.  I had no luck with the former, and the latter will always show the page momentarily before reloading the page.

**Reply by ryansolid** [reply]
Yeah this is just a bug. The value is invalidated by a mutation which means it shouldn't be in even the back cache.. We are more permissive in the back forward cache on time but invalidated data shouldn't get a free pass.

**Reply by JacobSNGoodwin** [reply]
Glad to know. I'm still figuring out the best way to do things and trying to learn, so it's totally possible I just missed a more logical way to handle this.

**Reply by ryansolid** [reply]
Yeah this was just an oversight on my part, some refactoring stuff.. I put the check in the wrong part of the conditional (it was in one of the or clauses instead of the and clause). I will get this out in the next release. https://github.com/solidjs/solid-router/commit/bceb358e28aa7e3e12ecf69fbf964f128d52c6c6
